specifyweb/frontend/js_src/lib/components/Statistics/definitions.ts

```diff
@@ -0,0 +1,36 @@
+import { State } from 'typesafe-reducer';
+import { statsSpec } from '../Statistics/StatsSpec';
+import { IR, RA } from '../../utils/types';
+import { statsText } from '../../localization/stats';
+
+type CustomStat = State<
+  'CustomStat',
+  {
+    readonly queryId: number;
+  }
+>;
+type DefaultStat = State<
+  'DefaultStat',
+  {
+    readonly categoryName: keyof typeof statsSpec;
+    readonly itemName: string;
+  }
+>;
+
+export type StatLayout = IR<IR<RA<CustomStat | DefaultStat>>>;
```

Comment:
Could you modify the type to the following?

```ts
export type StatLayout = IR<{
  readonly categories: IR<RA<CustomStat | DefaultStat>>;
}>;
```

This way if in the future would need to add some configuration options for a page (in addition to `IR<RA<CustomStat | DefaultStat>>`), it would be easy to extend the type. If we don't do this now, any new page options would cause ambiguity between option name and page label